assets/src/edit-story/components/autoSaveHandler/index.js

```diff
@@ -29,12 +29,17 @@ function AutoSaveHandler() {
   const {
     state: { hasNewChanges },
   } = useHistory();
-  const {
-    state: {
-      story: { status },
-    },
-    actions: { saveStory },
-  } = useStory();
+  const { status, saveStory } = useStory((state) => {
+    const {
+      state: {
+        // eslint-disable-next-line no-shadow
```

Comment:
These eslints will get very old very soon. Any way to deal with this automatically?

/cc @swissspidy for any ideas? Doesn't necessarily have to block this PR, but would be very good to see if we can resolve.

Alternative to this would be to extract this function into a top-level `const storySelector = ({...}) => {}` function. Would that help with lints? If so, I think that could have some cons and pros. But I feel like i'd rather keep it inline if we can do it well.

Reply:
Agreed. We should come up with a solution to improve this.

The rule itself is valuable to prevent headaches, so I'd suggest finding a workaround.

In other areas I see usage like this, and it works well especially if only a few items are selected:

```js
const { status, saveStory } = useStory((state) => ({
  status: state.state.status,
  saveStory: state.actions.saveStory,
}));
```

Alternatively, maybe our `objectPick` utility could be useful?

```js
const {
  currentPage,
  selectedElementIds,
  toggleElementInSelection,
  setSelectedElementsById,
} = useStory((state) => ({
  ...objectPick(state.state, ['currentPage', 'selectedElementIds']),
  ...objectPick(state.actions, ['toggleElementInSelection', 'setSelectedElementsById'])
}));
```

For comparison, that's half the lines of what's currently in `assets/src/edit-story/components/canvas/canvasProvider.js`:

```js
const {
  currentPage,
  selectedElementIds,
  toggleElementInSelection,
  setSelectedElementsById,
} = useStory(
  ({
    // eslint-disable-next-line no-shadow
    state: { currentPage, selectedElementIds },
    // eslint-disable-next-line no-shadow
    actions: { toggleElementInSelection, setSelectedElementsById },
  }) => {
    return {
      currentPage,
      selectedElementIds,
      toggleElementInSelection,
      setSelectedElementsById,
    };
  }
);
```

Reply:
Not a big fun of `objectPick` tbh. Here's my thinking here: the shadowing here happens in this form:

```
const {prop1, prop2} = functionToProvideProps(({prop1, prop2}) => {...})
```

I.e. an inline call shadows the top-level props it itself produces. IMHO this is a pretty justified case for shadowing. So I'm wondering if there might be some settings/customization available to avoid it. I'd even argue that there's no true prop shadowing here with `const` because until the `functionToProvideProps` returns the top-level `prop1` and `prop2` are actually not defined.

Reply:
@swissspidy wdyt if we tried `hoist: never` here? Not sure it fits 100%, but sounds like it might.

Reply:
Unfortunately changing the `hoist` option doesn't do anything, just tried.

To get to an agreement I propose the following:

1. Turn the rule's severity from `error` to `warn` for the time being
1. Report this to ESLint and maybe even contribute a fix (https://github.com/eslint/eslint/issues/12687 might cover this already)

Reply:
This would work for me. And yes, I definitely think that https://github.com/eslint/eslint/issues/12687 solves the issue.

Reply:
Updated to warn.